jenkins-scripts/lib/check_graphic_card.bash

```diff
@@ -3,16 +3,19 @@ GRAPHIC_CARD_PKG=""
 
 export_display_variable()
 {
-    # Hack to found the current display (if available) two steps:
-    # Check for /tmp/.X11-unix/ socket and check if the process is running
-    for i in $(ls /tmp/.X11-unix/ | sed -e 's@^X@:@')
+    # Check for an active X11 display socket.
+    for i in $(find /tmp/.X11-unix -type s)
     do
-      # grep can fail so let's disable the fail or error during its call
-      set +e
-      ps aux | grep bin/X.*$i | grep -v grep
-      set -e
-      if [ $? -eq 0 ] ; then
-	export DISPLAY=$i
+      # If a process is running with the open socket then
+      # lsof will exit successfully.
+      if lsof -Fp $i ; then
```

Comment:
With lsof we can "filter" socket files by whether or not an active process has each open rather than pattern matching all running processes to string match on a likely candidate.

Using it in the conditional directly also saves us having to bump around `set +e/-e`.